pkg/sidecars/server/handlers.go

```diff
@@ -112,20 +120,31 @@ func (s *Server) readWithRetry(ctx context.Context, channel string) (brokerMsg [
 	}
 }
 
-func (s *Server) writeWithRetry(ctx context.Context, channel string, data []byte) (status int, err error) {
+func (s *Server) writeWithRetry(
+	ctx context.Context,
+	channel string,
+	data []byte,
+) (status int, err error) {
 	var resp *http.Response
 	for i := 0; i <= maxBrokerRetries; i++ {
 		writeAddr := fmt.Sprintf("%s/%s", s.outAddr, channel)
 		logger.Debug("writing with retry",
 			zap.Any("addr", writeAddr),
 			zap.Any("write conter", i))
 
-		resp, err = s.client.Post(writeAddr, "application/octet-stream", bytes.NewBuffer(data))
+		resp, err = s.client.Post(
+			writeAddr,
+			"application/octet-stream",
+			bytes.NewBuffer(data),
+		)
+		if resp != nil {
+			defer resp.Body.Close()
+		}
+
```

Comment:
This is the modification, the rest is just a formatted that i ran while adding this